tools/nodejs_api/src_cpp/node_database.cpp

```diff
@@ -0,0 +1,69 @@
+#include "node_database.h"
+
+#include "main/kuzu.h"
+
+using namespace kuzu::main;
+
+Napi::FunctionReference NodeDatabase::constructor;
+
+Napi::Object NodeDatabase::Init(Napi::Env env, Napi::Object exports) {
+    Napi::HandleScope scope(env);
+
+    Napi::Function t = DefineClass(env, "NodeDatabase", {
+          InstanceMethod("resizeBufferManager", &NodeDatabase::ResizeBufferManager),
+      });
+
+    constructor = Napi::Persistent(t);
+    constructor.SuppressDestruct();
+
+    exports.Set("NodeDatabase", t);
+    return exports;
+}
+
+
+NodeDatabase::NodeDatabase(const Napi::CallbackInfo& info) : Napi::ObjectWrap<NodeDatabase>(info)  {
+    Napi::Env env = info.Env();
+    Napi::HandleScope scope(env);
+
+    if (info.Length()!=2) {
+        Napi::TypeError::New(env, "Need database config string and buffer manager size").ThrowAsJavaScriptException();
+    }
+    if (!info[0].IsString()) {
+        Napi::TypeError::New(env, "Database config parameter must be a string").ThrowAsJavaScriptException();
+    }
+    if (!info[1].IsNumber()) {
+        Napi::TypeError::New(env, "Database buffer manager size must be an int_64").ThrowAsJavaScriptException();
+    }
+
+    std::string databaseConfigString = info[0].ToString().Utf8Value().c_str();
```

Comment:
You should not cast the value to `c_str`. The cpp class supports UTF-8 string (non-English character). Casting it to `c_str` will break this functionality.